### PR TITLE
Shader module descriptor

### LINKED ISSUE #665: viewportCount set to 0 instead of 1, failing validation on Vulkan backend
I am not using multiview features, but it seems that the viewport count is incorrectly set at pipeline creation:

```
[0.110121 ERROR]()(no module): 
VALIDATION [VUID-VkPipelineViewportStateCreateInfo-viewportCount-01216 (-376016417)] : Validation Error: [ VUID-VkPipelineViewportStateCreateInfo-viewportCount-01216 ] Object 0: handle = 0x5628bad9a280, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xe99671df | vkCreateGraphicsPipelines: The VkPhysicalDeviceFeatures::multiViewport feature is disabled, but pCreateInfos[0].pViewportState->viewportCount (=0) is not 1. The Vulkan spec states: If the multiple viewports feature is not enabled, viewportCount must not be greater than 1 (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkPipelineViewportStateCreateInfo-viewportCount-01216)
object info: (type: DEVICE, hndl: 94732933505664)
```

Attaching `vulkaninfo`, `vulkan trace` and `wgpu trace`
[vkinfo.txt](https://github.com/gfx-rs/wgpu-rs/files/5646967/vkinfo.txt)
[trace.tar.gz](https://github.com/gfx-rs/wgpu-rs/files/5646968/trace.tar.gz)
[vk_apidump.log](https://github.com/gfx-rs/wgpu-rs/files/5646969/vk_apidump.log)

### PR DESCRIPTION
**Pull request by kvark**
Depends on https://github.com/gfx-rs/wgpu/pull/1071
Fixes #665
Based on #662

### CODE REVIEW

**Comment by kvark** [comment]
I imagine this may become a bit of a problem when rust-gpu decides that it can provide a Naga Module directly to wgpu-rs.
Not sure what to do with it, we need a separate discussion.

File: src/lib.rs [hunk 1]
```diff
@@ -687,6 +687,16 @@ pub enum ShaderModuleSource<'a> {
...
 }
 
+/// Descriptor for a shader module.
+pub struct ShaderModuleDescriptor<'a> {
+    /// Debug label of the shader module. This will show up in graphics debuggers for easy identification.
+    pub label: Label<'a>,
+    /// Source code for the shader.
+    pub source: ShaderSource<'a>,
+    /// Experimental translation path attempts to avoid SPIR-V and work with Naga IR directly.
+    pub experimental_translation: bool,
```
Comment (grovesNL) [inline src/lib.rs:697]: If we're mostly using this internally, then another option is to enable this through an environment variable. Then we wouldn't need to change the API later when we're ready to remove the SPIR-V path

**Review by grovesNL — APPROVED** [review]

**Review by cwfitzgerald — APPROVED** [review]
LGTM!

**Comment by kvark** [comment]
Bors r=cwfitzgerald,grovesNL

**Pull request merged**